exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JsonBatchReader.java

```diff
@@ -0,0 +1,149 @@
+import org.apache.drill.exec.store.dfs.DrillFileSystem;
+import org.apache.drill.exec.store.easy.json.parser.JsonLoaderImpl;
+import org.apache.drill.exec.store.easy.json.parser.JsonLoaderImpl.JsonOptions;
+import org.apache.drill.exec.store.easy.json.parser.JsonLoaderImpl.TypeNegotiator;
+import org.apache.hadoop.mapred.FileSplit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JsonBatchReader implements ManagedReader<FileSchemaNegotiator> {
+
+  private static final Logger logger = LoggerFactory.getLogger(JsonBatchReader.class);
+
+  private DrillFileSystem fileSystem;
+  private FileSplit split;
+  private final JsonOptions options;
+  private JsonLoader jsonLoader;
+  private InputStream stream;
+
+  private RowSetLoader tableLoader;
+
+  public JsonBatchReader(JsonOptions options) {
+   this.options = options == null ? new JsonOptions() : options;
+  }
+
+  @Override
+  public boolean open(FileSchemaNegotiator negotiator) {
+    this.fileSystem = negotiator.fileSystem();
+    this.split = negotiator.split();
+    OperatorContext opContext = negotiator.context();
+    OptionSet optionMgr = opContext.getFragmentContext().getOptions();
+    Object embeddedContent = null;
+    options.allTextMode = embeddedContent == null && optionMgr.getBoolean(ExecConstants.JSON_ALL_TEXT_MODE);
+    options.readNumbersAsDouble = embeddedContent == null && optionMgr.getBoolean(ExecConstants.JSON_READ_NUMBERS_AS_DOUBLE);
+    options.unionEnabled = embeddedContent == null && optionMgr.getBoolean(ExecConstants.ENABLE_UNION_TYPE_KEY);
+    options.skipMalformedRecords = optionMgr.getBoolean(ExecConstants.JSON_READER_SKIP_INVALID_RECORDS_FLAG);
+    // Printing of malformed records is always enabled.
+//    options.printSkippedMalformedJSONRecordLineNumber = optionMgr.getBoolean(ExecConstants.JSON_READER_PRINT_INVALID_RECORDS_LINE_NOS_FLAG);
+    options.allowNanInf = true;
+
+    try {
+      stream = fileSystem.openPossiblyCompressedStream(split.getPath());
+    } catch (IOException e) {
+      throw UserException
+          .dataReadError(e)
+          .addContext("Failure to open JSON file", split.getPath().toString())
+          .build(logger);
+    }
+    final ResultSetLoader rsLoader = negotiator.build();
+    tableLoader = rsLoader.writer();
+    RowSetLoader rootWriter = tableLoader;
+
+    // Bind the type negotiator that will resolve ambiguous types
+    // using information from any previous readers in this scan.
+
+    options.typeNegotiator = new TypeNegotiator() {
+      @Override
+      public MajorType typeOf(List<String> path) {
+        ResultVectorCache cache = rsLoader.vectorCache();
+        for (int i = 0; i < path.size() - 1; i++) {
+          cache = cache.childCache(path.get(i));
+        }
+        return cache.getType(path.get(path.size()-1));
+      }
+    };
+
+    // Create the JSON loader (high-level parser).
+
+    jsonLoader = new JsonLoaderImpl(stream, rootWriter, options);
```

Comment:
@paul-rogers 
Would it be possible for you to add some logic here so that the JSONBatchReader can accept an `InputStream` as well as file input?  This could be useful for other storage plugins that output JSON.